dwave/plugins/torch/samplers/block_spin_sampler.py

```diff
@@ -0,0 +1,348 @@
+            "Gibbs" or "Metropolis".
+    """
+
+    def to(self, device: DeviceLikeType) -> None:
+        """Moves sampler components to the target device in-place.
+
+        If the device is "meta", then the random number generator ``BlockSpinSampler._rng`` (RNG)
+        will not be modified at all. For all other devices, all fields ``_x``, ``_zeros``,
+        ``_schedule``, ``_partition``, ``_padded_adjacencies``, and ``_padded_adjacencies_weight``
+        will be moved to the target device. Importantly, the RNG's device is relayed by the
+        following procedure:
+        1. Draw a random integer between 0 (inclusive) and 2**60 (exclusive) with the current
+           generator as a new seed ``s``.
+        2. Create a new generator on the target device.
+        3. Set the new generator's seed as ``s``.
+
+        Args:
+            device (DeviceLikeType): The target device.
+        """
+        self._x = self._x.to(device)
+        self._zeros = self._zeros.to(device)
+        self._schedule = self._schedule.to(device)
+        self._partition = self._partition.to(device)
+        self._padded_adjacencies = self._padded_adjacencies.to(device)
+        self._padded_adjacencies_weight = self._padded_adjacencies_weight.to(device)
+        if device != "meta":
+            rng = torch.Generator(device)
+            rng.manual_seed(torch.randint(0, 2**60, (1,), generator=self._rng).item())
+            self._rng = rng
+
+    def __init__(self, grbm: GRBM, crayon: Callable[[Hashable], Hashable], num_chains: int,
+                 schedule: Iterable[float],
+                 proposal_acceptance_criteria: Literal["Gibbs", "Metropolis"] = "Gibbs",
+                 initial_states: torch.Tensor | None = None,
+                 seed: int | None = None):
+        super().__init__()
+
+        if not isinstance(num_chains, int) or num_chains < 1:
+            raise NonPositiveIntegerError(
+                "Number of reads should be a positive integer."
+            )
+
+        self._proposal_acceptance_criteria = proposal_acceptance_criteria.title()
+        if self._proposal_acceptance_criteria not in {"Gibbs", "Metropolis"}:
+            raise InvalidProposalAcceptanceCriteriaError(
+                f'Proposal acceptance criterion should be one of "Gibbs" or "Metropolis"'
+            )
+
+        self._grbm = grbm
+        self._crayon = crayon
+        if not self._valid_crayon():
+            raise InvalidColouringError(
+                "``crayon`` is not a valid colouring of ``grbm``. "
+                "At least one edge has vertices of the same colour."
+            )
+
+        self._partition = self._get_partition()
+        self._padded_adjacencies, self._padded_adjacencies_weight = self._get_adjacencies()
+
+        self._rng = None
+        if seed is not None:
+            self._rng = torch.Generator()
+            self._rng.manual_seed(seed)
```

Comment:
(minor) IMO slightly clearer.
```suggestion
        if seed is not None:
            self._rng = torch.Generator()
            self._rng.manual_seed(seed)
        else:
            self._rng = None
```

Reply:
I think the original is cleaner 😝